plugin/pkg/managedseed/validator/admission.go

```diff
@@ -550,3 +558,24 @@ func (v *ManagedSeed) getSecretBinding(namespace, name string) (*gardencore.Secr
 func (v *ManagedSeed) getSecrets(namespace string, selector labels.Selector) ([]*corev1.Secret, error) {
 	return v.secretLister.Secrets(namespace).List(selector)
 }
+
+func addSeedSecretRefAnnotations(managedSeed *seedmanagement.ManagedSeed) error {
+	if managedSeed.Spec.Gardenlet != nil &&
+		managedSeed.Spec.Gardenlet.Config != nil {
+
+		// Convert gardenlet config to an internal version
+		gardenletConfig, err := gardenlethelper.ConvertGardenletConfiguration(managedSeed.Spec.Gardenlet.Config)
+		if err != nil {
+			return apierrors.NewInternalError(fmt.Errorf("could not convert config: %v", err))
+		}
+
+		if gardenletConfig.SeedConfig != nil {
+			if secretRef := gardenletConfig.SeedConfig.Spec.SecretRef; secretRef != nil {
+				metav1.SetMetaDataAnnotation(&managedSeed.ObjectMeta, seedmanagementv1alpha1constants.AnnotationSeedSecretName, secretRef.Name)
+				metav1.SetMetaDataAnnotation(&managedSeed.ObjectMeta, seedmanagementv1alpha1constants.AnnotationSeedSecretNamespace, secretRef.Namespace)
```

Comment:
Is there a mechanism to clean up the annotations after Seed spec.secretRef is unset?

Reply:
It's cleaned up after the secret is deleted,  See this commit, https://github.com/gardener/gardener/pull/8039/commits/3a4241deb0281252d4ce3df6f96dda614623df80